subprojects/platform-jvm/src/main/java/org/gradle/api/java/archives/internal/DefaultManifest.java

```diff
@@ -145,7 +145,9 @@ private static void addMainAttributesToJavaManifest(org.gradle.api.java.archives
         for (Map.Entry<String, Object> entry : gradleManifest.getAttributes().entrySet()) {
             String mainAttributeName = entry.getKey();
             String mainAttributeValue = resolveValueToString(entry.getValue());
-            javaManifest.getMainAttributes().putValue(mainAttributeName, mainAttributeValue);
+            if (mainAttributeValue != null) {
+                javaManifest.getMainAttributes().putValue(mainAttributeName, mainAttributeValue);
+            }
```

Comment:
Besides only fixing the bug once, I think it would help with readability to refactor this and pull out the common code that transforms the attributes. This is exactly the same as lines 159-163, just different names :)

```
attribts.stream().filter(value != null).collect(toMap(..))
```

Reply:
Extracted the common method indeed.
However decided against the functional form as it is messy:
```
        attributes.entrySet().stream().peek(entry -> entry.setValue(resolveValueToString(entry.getValue())))
            .filter(entry -> entry.getValue() != null)
            .forEach(entry -> targetAttributes.putValue(entry.getKey(), entry.getValue().toString()));
```
The above is the shortest form. A proper form with `map` is annoying because of having to deal with Map.Entry instance creation. So I left the imperative code instead.